packages/lib/src/vis/shared/ViewportCenterer.tsx

```diff
@@ -0,0 +1,29 @@
+import { useFrame, useThree } from '@react-three/fiber';
+import { useEffect, useRef } from 'react';
+import type { Vector2 } from 'three';
+
+import { useAxisSystemContext } from './AxisSystemContext';
+import { useMoveCameraTo } from './hooks';
+
+function ViewportCenterer() {
+  const { dataToWorld, worldToData } = useAxisSystemContext();
+  const viewportCenter = useRef<Vector2>();
+  const { position } = useThree((state) => state.camera);
+
+  const moveCameraTo = useMoveCameraTo();
+
+  useFrame(() => {
+    viewportCenter.current = worldToData(position);
+  });
+
+  useEffect(() => {
+    if (viewportCenter.current) {
+      // On resize, move camera to the latest saved viewport center coordinates
+      const { x, y } = dataToWorld(viewportCenter.current);
+      moveCameraTo(x, y);
+    }
+  }, [viewportCenter, moveCameraTo, dataToWorld, position.x, position.y]);
+  return <mesh />;
```

Comment:
I was unsure of what kind of component to put here. As everything happens in hooks, we might instead put the logic in a custom hook and call this hook in `AxisSystem` instead ? :shrug:

Reply:
Can you return `null`? I think it's good to have a component that can be rendered explicitly from `VisCanvas`.

Reply:
`null` seems to do the trick